components/right-sidebar/SocialMediaModal.tsx

```diff
@@ -0,0 +1,178 @@
+"use client";
+
+import { useEffect, useRef } from "react";
+import { Loader2, X } from "lucide-react";
+import {
+  SubmitHandler,
+  useForm,
+  Controller,
+  FieldError,
+} from "react-hook-form";
+import toast from "react-hot-toast";
+import { zodResolver } from "@hookform/resolvers/zod";
+
+import { updateUser } from "@/lib/actions/user.actions";
+import {
+  ISocialMediaSchema,
+  IUserSchema,
+  SocialMediaSchema,
+} from "@/lib/validations/UserSchema";
+import { Social, User } from "@prisma/client";
+import Input from "../shared/ui/Input";
+import { socialMediaIconList } from "@/lib/constants/socialMediaList";
+import Button from "../shared/ui/Button";
+
+const SocialMediaModal = ({
+  user,
+  onClose,
+}: {
+  user: User & { socialMedia?: Social[] };
+  onClose: () => void;
+}) => {
+  const ref = useRef(null);
+
+  useEffect(() => {
```

Comment:
Was meaning to add this earlier as well - hope it isn't overkill to have both of these!